web/modules/custom/joinup_discussion/tests/src/Functional/DiscussionWorkflowTest.php

```diff
@@ -235,299 +167,163 @@ protected function isPublishedState($state) {
-          ['update', 'userOgFacilitator', TRUE],
-          ['update', 'userOgAdministrator', FALSE],
-          ['delete', 'userAnonymous', FALSE],
-          ['delete', 'userAuthenticated', FALSE],
-          ['delete', 'userModerator', TRUE],
-          ['delete', 'userOgFacilitator', TRUE],
-          ['delete', 'userOgAdministrator', FALSE],
+          'userAuthenticated' => [],
+          'userOwner' => [],
+          'userOgMember' => [],
+          'userOgFacilitator' => [
+            'approve_report',
+          ],
+          'userModerator' => [
+            'approve_report',
+          ],
         ],
         'proposed' => [
-          ['view', 'userAnonymous', FALSE],
-          ['view', 'userAuthenticated', FALSE],
-          ['view', 'userModerator', TRUE],
-          ['view', 'userOgFacilitator', TRUE],
-          ['view', 'userOgAdministrator', FALSE],
-          ['update', 'userAnonymous', FALSE],
-          ['update', 'userAuthenticated', FALSE],
-          ['update', 'userModerator', TRUE],
-          ['update', 'userOgFacilitator', TRUE],
-          ['update', 'userOgAdministrator', FALSE],
-          ['delete', 'userAnonymous', FALSE],
-          ['delete', 'userAuthenticated', FALSE],
-          ['delete', 'userModerator', TRUE],
-          ['delete', 'userOgFacilitator', TRUE],
-          ['delete', 'userOgAdministrator', FALSE],
+          'userAuthenticated' => [],
+          'userOwner' => [
+            'update_proposed',
+          ],
+          'userOgMember' => [],
+          'userOgFacilitator' => [
+            'update_proposed',
+            'approve_proposed',
+          ],
+          'userModerator' => [
+            'update_proposed',
+            'approve_proposed',
+          ],
         ],
+        // Once the node is in archived state, no actions can be taken anymore.
         'archived' => [
-          ['view', 'userAnonymous', TRUE],
-          ['view', 'userAuthenticated', TRUE],
-          ['view', 'userModerator', TRUE],
-          ['view', 'userOgFacilitator', TRUE],
-          ['view', 'userOgAdministrator', TRUE],
-          ['update', 'userAnonymous', FALSE],
-          ['update', 'userAuthenticated', FALSE],
-          ['update', 'userModerator', TRUE],
-          ['update', 'userOgFacilitator', FALSE],
-          ['update', 'userOgAdministrator', FALSE],
-          ['delete', 'userAnonymous', FALSE],
-          ['delete', 'userAuthenticated', FALSE],
-          ['delete', 'userModerator', TRUE],
-          ['delete', 'userOgFacilitator', TRUE],
-          ['delete', 'userOgAdministrator', FALSE],
+          'userAuthenticated' => [],
+          'userOwner' => [],
+          'userOgMember' => [],
+          'userOgFacilitator' => [],
+          'userModerator' => [],
         ],
       ],
     ];
+
+    $parent_access_array = [];
+    $return_array = [];
+    $e_library_states = $this->getElibraryStates();
+
+    foreach ($access_array as $moderation_state => $moderation_data) {
```

Comment:
Drop a comment explaining that since the permissions are the same for parents, parent state, etc you wrote the array once and then copy it over all the possible cases.